Index: node/tools/gyp/pylib/gyp/generator/make.py
===================================================================
--- node.orig/tools/gyp/pylib/gyp/generator/make.py
+++ node/tools/gyp/pylib/gyp/generator/make.py
@@ -308,7 +308,7 @@ CXX.target ?= %(CXX.target)s
 CXXFLAGS.target ?= $(CPPFLAGS) $(CXXFLAGS)
 LINK.target ?= %(LINK.target)s
 LDFLAGS.target ?= $(LDFLAGS)
-AR.target ?= $(AR)
+AR.target ?= %(AR.target)s
 
 # C++ apps need to be linked with g++.
 LINK ?= $(CXX.target)
